### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by evanchooly**
updates for document validation and partial indexing

### CODE REVIEW

File: docs/reference/content/guides/annotations.md [hunk 1]
```diff
@@ -32,6 +32,16 @@ The [`@IndexOptions`]({{< apiref "org/mongodb/morphia/annotations/IndexOptions"
 definition.  This annotation replaces the fields found directly on the `@Index` annotation.  This annotation was added to ensure that index
 options are consistent across the various index definition approaches.
 
+#### Collation
+The [`@Collation`]({{< apiref "org/mongodb/morphia/annotations/Collation" >}}) annotation defines the options to apply to the collation of
```
Comment (kay-kim) [inline docs/reference/content/guides/annotations.md]: ... annotation defines the collation options to apply to the index definition.

File: docs/reference/content/guides/annotations.md [hunk 2]
```diff
@@ -32,6 +32,16 @@ The [`@IndexOptions`]({{< apiref "org/mongodb/morphia/annotations/IndexOptions"
...
+an index definition.  These collation definitions can significantly affect how documents are matched.  In addition to defining a collation
+as part of an index, a collation can be specified as part of a query as well.  The Options classes provide facilities for specifying a
```
Comment (kay-kim) [inline docs/reference/content/guides/annotations.md]: Q: With morphia, can you specify the collation option when creating a collection?

File: docs/reference/content/guides/annotations.md [hunk 3]
```diff
@@ -32,6 +32,16 @@ The [`@IndexOptions`]({{< apiref "org/mongodb/morphia/annotations/IndexOptions"
 definition.  This annotation replaces the fields found directly on the `@Index` annotation.  This annotation was added to ensure that index
 options are consistent across the various index definition approaches.
 
+#### Collation
+The [`@Collation`]({{< apiref "org/mongodb/morphia/annotations/Collation" >}}) annotation defines the options to apply to the collation of
+an index definition.  These collation definitions can significantly affect how documents are matched.  In addition to defining a collation
```
Comment (kay-kim) [inline docs/reference/content/guides/annotations.md]: Should we be more explicit than "can significantly affect how documents are matched?"  since it would only impact string comparisons?

File: docs/reference/content/guides/annotations.md [hunk 4]
```diff
@@ -32,6 +32,16 @@ The [`@IndexOptions`]({{< apiref "org/mongodb/morphia/annotations/IndexOptions"
...
+as part of an index, a collation can be specified as part of a query as well.  The Options classes provide facilities for specifying a 
+specific collation to be used for any given operation.  This collation does not have to match the one defined on the index but will, of
```
Comment (kay-kim) [inline docs/reference/content/guides/annotations.md]: I don't think it's "any given operation" since that makes it sound like it's for all operations.

File: docs/reference/content/guides/annotations.md [hunk 5]
```diff
@@ -32,6 +32,16 @@ The [`@IndexOptions`]({{< apiref "org/mongodb/morphia/annotations/IndexOptions"
...
+course, be faster if it does.  See [`CountOptions`]({{< apiref "org/mongodb/morphia/CountOptions" >}}),
```
Comment (kay-kim) [inline docs/reference/content/guides/annotations.md]: Is it more that an index with a collation can support  operations that specifies a collation if the collations match? 

I recognize that I'm oversimplifying in cases of compound indexes where non-string keys are the index prefixes and operations specify a filter of ( { x: 5, y: 2, somestring: "boo" } ) or whatever.

File: docs/reference/content/guides/annotations.md [hunk 6]
```diff
@@ -66,6 +76,29 @@ In contrast to `@Reference` where a nested Java reference ends up as a separate
...
+public class SomeEntity {
+    ...
+    private int number;
+    ...
+}
+```
+
+This validation is applied to every insert and update in the mapped collection.  In some cases, it's necessary to bypass this validation. To
+ this end, the various operations on [`Datastore`]({{< apiref "org/mongodb/morphia/Datastore" >}}) and 
+[`AdvancedDatastore`]({{< apiref "org/mongodb/morphia/AdvancedDatastore" >}}) take Options classes.  There are varying options to be set on
```
Comment (kay-kim) [inline docs/reference/content/guides/annotations.md]: Should be just collapse the three sentences to:
Various [`Datastore`] and [`AdvancedDatastore`] operations can bypass this validation via their Options classes. For these operations, specify the `bypassDocumentValidation` option  to disable document validation for a specific operation.

File: docs/reference/content/guides/annotations.md (hunk 1 above)
Comment (evanchooly) [inline docs/reference/content/guides/annotations.md]: done

File: docs/reference/content/guides/annotations.md (hunk 3 above)
Comment (evanchooly) [inline docs/reference/content/guides/annotations.md]: I think i'll just remove that line in favor of a link to the full collation docs on the main docs site.

File: docs/reference/content/guides/annotations.md (hunk 1 above)
Comment (evanchooly) [inline docs/reference/content/guides/annotations.md]: done

File: docs/reference/content/guides/annotations.md (hunk 2 above)
Comment (evanchooly) [inline docs/reference/content/guides/annotations.md]: At the moment, no.  The only time morphia explicitly creates a collection is for document validation.

File: docs/reference/content/guides/annotations.md (hunk 6 above)
Comment (evanchooly) [inline docs/reference/content/guides/annotations.md]: done

File: morphia/src/test/java/org/mongodb/morphia/query/TestQuery.java [hunk 7]
```diff
@@ -1172,7 +1173,9 @@ public void run() {
             }
         }, 0, 500, TimeUnit.MILLISECONDS);
 
-        final Iterator<CappedPic> tail = query.tail();
+        final Iterator<CappedPic> tail = query
```
Comment (jyemin) [inline morphia/src/test/java/org/mongodb/morphia/query/TestQuery.java]: Why is this change in this patch?  It seems unrelated.

Comment (evanchooly) [inline morphia/src/test/java/org/mongodb/morphia/query/TestQuery.java]: It's a change I made to make sure when i update the docs that I do so with code that's correct.
